lib/scanner/bem-walk.js

```diff
@@ -1,31 +1,29 @@
-var bemWalk = require('bem-walk'),
-    q = require('q');
+var bemWalk = require('bem-walk');
 
 /**
  * @param {String[]} levels
  * @returns {Promise<Object[]>}
  */
 exports.walk = function(levels) {
     var walker = bemWalk(levels),
-        defer = q.defer(),
         techs = [];
 
-    walker.on('data', function(data) {
-        techs.push({
-            entity: data.entity,
-            level: data.level,
-            name: data.tech,
-            path: data.path
+    return new Promise((resolve, reject) => {
+        walker.on('data', function(data) {
+            techs.push({
+                entity: data.entity,
+                level: data.level,
+                name: data.tech,
+                path: data.path
+            });
         });
-    });
 
-    walker.on('error', function(err) {
-        defer.reject(err);
-    });
+        walker.on('error', function(err) {
+            reject(err);
+        });
 
-    walker.on('end', function() {
-        defer.resolve(techs);
+        walker.on('end', function() {
+            resolve(techs);
```

Comment:
If you wish to use ES6 need to replace this function with `() => resolve(techs)`